pkg/api/v1/router_server_components.go

```diff
@@ -136,15 +155,32 @@ func (r *Router) serverComponentsCreate(c *gin.Context) {
 		for _, versionedAttributes := range srvComponent.VersionedAttributes {
 			dbVersionedAttributes := versionedAttributes.toDBModel()
 			dbVersionedAttributes.ServerID = null.StringFrom(server.ID)
-			dbVersionedAttributes.ServerComponentID = null.StringFrom(srvComponent.UUID.String())
+			dbVersionedAttributes.ServerComponentID = null.StringFrom(dbSrvComponent.ID)
 
-			// insert true
 			err = dbSrvComponent.AddVersionedAttributes(c.Request.Context(), tx, true, dbVersionedAttributes)
 			if err != nil {
 				dbErrorResponse(c, err)
 				return
 			}
 		}
+
+		// insert versioned attributes
+		for _, attributes := range srvComponent.Attributes {
+			dbAttributes, err := attributes.toDBModel()
+			if err != nil {
+				dbErrorResponse(c, err)
+				return
+			}
+
+			dbAttributes.ServerID = null.StringFrom(server.ID)
```

Comment:
We shouldn't be attaching this to both the server ID and the server component ID. 

This is something I want to see refactored because as this codebase grows this won't be scalable. But the attribute should only be assigned to a single item or else it will be returned in lists it shouldn't be returned in.